format/FlightSql.proto

```diff
@@ -867,6 +867,153 @@ enum SqlSupportsConvert {
   SQL_CONVERT_VARCHAR = 19;
 }
 
+enum SqlDatetimeSubcode {
+  SQL_DATETIME_SUBCODE_UNKNOWN = 0;
+  SQL_DATETIME_SUBCODE_DATE = 1;
+  SQL_DATETIME_SUBCODE_TIME = 2;
+  SQL_DATETIME_SUBCODE_TIMESTAMP = 3;
+}
+
+enum SqlDataType {
+  SQL_CODE_UNKNOWN = 0;
+  SQL_CODE_YEAR = 1;
+  SQL_CODE_MONTH = 2;
+  SQL_CODE_DAY = 3;
+  SQL_CODE_HOUR = 4;
+  SQL_CODE_MINUTE = 5;
+  SQL_CODE_SECOND = 6;
+  SQL_CODE_YEAR_TO_MONTH = 7;
+  SQL_CODE_DAY_TO_HOUR = 8;
+  SQL_CODE_DAY_TO_MINUTE = 9;
+  SQL_CODE_DAY_TO_SECOND = 10;
+  SQL_CODE_HOUR_TO_MINUTE = 11;
+  SQL_CODE_HOUR_TO_SECOND = 12;
+  SQL_CODE_MINUTE_TO_SECOND = 13;
+  SQL_INTERVAL_YEAR = 101;
+  SQL_INTERVAL_MONTH = 102;
+  SQL_INTERVAL_DAY = 103;
+  SQL_INTERVAL_HOUR = 104;
+  SQL_INTERVAL_MINUTE = 105;
+  SQL_INTERVAL_SECOND = 106;
+  SQL_INTERVAL_YEAR_TO_MONTH = 107;
+  SQL_INTERVAL_DAY_TO_HOUR = 108;
+  SQL_INTERVAL_DAY_TO_MINUTE = 109;
+  SQL_INTERVAL_DAY_TO_SECOND = 110;
+  SQL_INTERVAL_HOUR_TO_MINUTE = 111;
+  SQL_INTERVAL_HOUR_TO_SECOND = 112;
+  SQL_INTERVAL_MINUTE_TO_SECOND = 113;
```

Comment:
And now I'm looking at this and trying to cross reference it with the ODBC docs -  I thought the point of `sql_data_type` is that it only returned a generic INTERVAL value, so why is this enum referenced by `sql_data_type` below and not `data_type`, and why do we have all the individual interval/datetime types?

Reply:
It's very confusing. There are 3 fields involved:
data_type - this holds the exact type, such as SQL_INTERVAL_DAY
sql_data_type - this will hold a generic interval or datetime SQL type such as SQL_INTERVAL
sql_datetime_sub - this will hold a subcode for the interval/datetime type which may be a different value than the data_type.

Reply:
Ah in that case - we should have constants for just SQL_INTERVAL and so on, right? And these constants should be used only by data_type?